src/sagemaker_training/mpi.py

```diff
@@ -341,6 +341,18 @@ def _create_command(self):
             command.extend(["-x", "FI_PROVIDER=efa"])
             # Use simple protocol to handle the out-of-order data delivery from EFA
             command.extend(["-x", "NCCL_PROTO=simple"])
+            # Further set NCCL_ALGO for modelparallel job to avoid hang, starting from torch2.0.0
+            env = environment.Environment()
+            if env.is_modelparallel_enabled:
+                try:
+                    import torch
```

Comment:
generally we don't recommend import a specific framework in training toolkit, double check with @satishpasumarthi if this is appropriate

Reply:
Updated with changes in SMPTTT